core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -699,6 +701,10 @@ public BlockReader createBlockReader(BlockReadRequest request) throws
         return createUfsBlockReader(request.getSessionId(), request.getId(), request.getStart(),
             request.isPositionShort(), request.getOpenUfsBlockOptions());
       } catch (Exception e) {
+        // createUfsBlockReader() wraps UFS exception as IOException
+        if (e.getCause() instanceof FileNotFoundException) {
+          throw new NotFoundException(e.getMessage(), e.getCause());
+        }
```

Comment:
Translated version of https://github.com/Alluxio/alluxio/blob/6031ffee3a2eb8db859ba078d3a2dabde45dca35/core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java#L184-L186

since `branch-2.8` does not have the `BlockWorker` refactor from https://github.com/Alluxio/alluxio/pull/15719 yet.

Reply:
Did not add the corresponding DefaultBlockWorkerTest: https://github.com/Alluxio/alluxio/blob/bf34b0e2efb83a1eef9677dc6f24830b31d84dec/core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java#L556-L572

It encountered NPE and is likely dependent on the following two PRs:
- https://github.com/Alluxio/alluxio/pull/15791
  - Likely dependant on https://github.com/Alluxio/alluxio/pull/15719 which is too large to include for this fix
- https://github.com/Alluxio/alluxio/pull/15958

Reply:
Yes, I think this change is just for the test code.  It might be very difficult to reach here for production traffic pattern